src/ApparentHorizons/ChangeCenterOfStrahlkorper.cpp

```diff
@@ -0,0 +1,158 @@
+  const DataVector sin_theta = sin(new_theta_phi[0]);
+  get<0>(r_hat_l) = sin_theta * cos(new_theta_phi[1]);
+  get<1>(r_hat_l) = sin_theta * sin(new_theta_phi[1]);
+  get<2>(r_hat_l) = cos(new_theta_phi[0]);
+  return r_hat_l;
+}
+
+// Work function called by the two interface functions.
+template <typename Frame>
+void change_expansion_center(
+    const gsl::not_null<Strahlkorper<Frame>*> strahlkorper,
+    const std::array<double, 3>& new_center,
+    const tnsr::i<DataVector, 3, Frame>& r_hat) noexcept {
+
+  // Get new_center minus old_center.
+  const auto& old_center = strahlkorper->center();
+  const std::array<double, 3> center_difference{
+      {new_center[0] - old_center[0], new_center[1] - old_center[1],
+       new_center[2] - old_center[2]}};
+
+  // For bracketing the root, get the min and max radius with respect
+  // to the new center.
+  const auto [r_min, r_max] =
+      [&center_difference, &r_hat, &strahlkorper ]() noexcept {
+    const auto radius_old = strahlkorper->ylm_spherepack().spec_to_phys(
+        strahlkorper->coefficients());
+    DataVector radius_new(radius_old.size(), 0.0);
+    for (size_t d = 0; d < 3; ++d) {
+      radius_new +=
+          square(r_hat.get(d) * radius_old - gsl::at(center_difference, d));
+    }
+    radius_new = sqrt(radius_new);
+    const auto minmax =
+        std::minmax_element(radius_new.begin(), radius_new.end());
+    return std::make_pair(*(minmax.first), *(minmax.second));
+  }
+  ();
+
+  // Find the coordinate radius of the surface, with respect to the
+  // new center, at each of the angular collocation points of the
+  // surface with respect to the new center. To do so, for each index
+  // 's' (corresponding to an angular collocation point), find the
+  // root 'r_new' that zeroes this lambda function.
+  const auto radius_function = [&center_difference, &r_hat, &strahlkorper ](
+      const double r_new, const size_t s) noexcept {
+    // Get cartesian coordinates of the point with respect to the old
+    // center.
+    const std::array<double, 3> x_old{
+        {r_new * get<0>(r_hat)[s] + center_difference[0],
+         r_new * get<1>(r_hat)[s] + center_difference[1],
+         r_new * get<2>(r_hat)[s] + center_difference[2]}};
+
+    // Find (r, theta, phi) of the same point with respect to the old_center.
+    const double r_old =
+        sqrt(square(x_old[0]) + square(x_old[1]) + square(x_old[2]));
+    const double theta_old = acos(x_old[2] / r_old);
+    const double phi_old = atan2(x_old[1], x_old[0]);
+
+    // Evaluate the radius of the surface on the old strahlkorper
+    // at theta_old, phi_old.
+    const double r_surf_old = strahlkorper->radius(theta_old, phi_old);
+
+    // If r_surf_old is r_old, then 'r_new' is on the surface.
+    return r_surf_old - r_old;
+  };
+
+  const auto radius_at_each_angle = RootFinder::toms748(
+      radius_function, make_with_value<DataVector>(get<0>(r_hat), r_min * 0.9),
```

Comment:
Should the 0.9 and 1.1 be hardcoded? If so, maybe there should be a single variable storing the 10% padding for the bracketing, so it's easy to change if needed later?